ui/js/graphs.js

```diff
@@ -764,11 +780,73 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
         $modalInstance.dismiss('cancel');
     };
 
+    $scope.testsToAdd = []; // tests in the "tests to add" list
     $scope.unselectedTestList = []; // tests in the "tests" list
     $scope.selectedTestSignatures = []; // tests in the "tests" list that have been selected by the user
-    $scope.testsToAdd = []; // tests in the "tests to add" list
     $scope.selectedTestsToAdd = []; // tests in the "to add" test list that have been selected by the user
 
+    var relateSeries = options.relatedSeries;
+    // if we choose to add all related platforms, firstly we need
+    // to obtain all the platform of this project, then pick out the
+    // specific test by platform and project.
+    var addRelatePlaform = function(relateSeries) {
+        PhSeries.getAllSeries(relateSeries.projectName,
```

Comment:
@wlach  How could I promise the code between line 793 - 805 has been executed before I return the length of $scope.testToAdd.  Otherwise, it always return undefined. I had try to use $q.all() to wrap them but doesn't work :(